compiler/packages/babel-plugin-react-compiler/src/Validation/ValidatePreservedManualMemoization.ts

```diff
@@ -403,19 +403,15 @@ class Visitor extends ReactiveFunctionVisitor<VisitorState> {
       state.manualMemoState.decls.add(lvalue.identifier.declarationId);
     }
 
-    const maybeDep = this.recordDepsInValue(value, state);
+    this.recordDepsInValue(value, state);
     if (lvalId != null) {
-      if (maybeDep != null) {
-        temporaries.set(lvalId, maybeDep);
-      } else if (isNamedLocal) {
-        temporaries.set(lvalId, {
-          root: {
-            kind: 'NamedLocal',
-            value: {...(instr.lvalue as Place)},
-          },
-          path: [],
-        });
-      }
+      temporaries.set(lvalId, {
+        root: {
+          kind: 'NamedLocal',
+          value: {...(instr.lvalue as Place)},
```

Comment:
nit: above we could refine on `lvalue != null` and then use `lvalue` here without `as`